pip/flatpak-pip-generator

```diff
@@ -169,11 +150,65 @@ with open(requirements_file, 'r') as req_file:
     use_hash = '--hash=' in req_file.read()
 
 python_version = '2' if opts.python2 else '3'
+devnull = subprocess.DEVNULL
+
+# Prefer to use Sdk as they bundle `git`.
+possible_runtimes = [
+    'org.freedesktop.Sdk',
+    'org.freedesktop.Platform',
+    'org.gnome.Sdk',
+    'org.gnome.Platform',
+    'org.kde.Sdk',
+    'org.kde.Platform',
+    # If multiple branches are available, the script won't select any.
+    'org.freedesktop.Sdk//19.08',
+    'org.freedesktop.Platform//19.08',
+]
+if opts.runtime:
+    runtime = opts.runtime
+else:
+    for flatpak in possible_runtimes:
+        try:
+            flatpak_info = ['flatpak', 'info', '--show-runtime'] + [flatpak]
+            if subprocess.run(flatpak_info, check=True, stdout=devnull, stderr=devnull):
+                runtime = flatpak
+                break
+        except subprocess.CalledProcessError:
+            pass
+    else:
+        if not opts.requirements:
+            # The temporal requirements file is still around.
+            try:
+                os.remove(requirements_file)
+            except FileNotFoundError:
+                pass
+        print('No flatpak runtime could be found, try specifying a flatpak or runtime version')
+        exit('e.g. "flatpak-pip-generator --runtime domain.developer.app-id//branch [packages]"')
+
+    print('Using {} as a sandbox'.format(runtime))
+
 if opts.python2:
     pip_executable = 'pip2'
 else:
     pip_executable = 'pip3'
 
+if opts.python2 and not opts.runtime:
+    print('WARNING: --python2 requires a flatpak or runtime that bundles pip2 to be specified with --runtime for this script to be run inside a sandbox')
```

Comment:
Since no supported runtime contains `python2` maybe we should be more explicit that you should specify an application and not just runtimes.

Reply:
Changed to "flatpak app", might be better to say application.